packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -118,3 +126,58 @@ export const useLocalVideoStartTrigger = (isLocalVideoAvailable: boolean, should
     }
   }, [shouldTransition, isButtonStatusSynced, isPreviewCameraOn, isLocalVideoAvailable, mediaGalleryHandlers]);
 };
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ */
+const useParticipantChangedAnnouncement = (): string => {
+  const adapter = useAdapter();
+  const locale = useLocale().strings.call;
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  /**
+   * state to track whether there is currently a timer set in the MediaGallery
+   */
+  const [timeoutState, setTimeoutState] = useState<ReturnType<typeof setTimeout>>();
+
+  const setParticipantEventString = (string: string): void => {
+    setAnnouncerString('');
+
+    if (timeoutState) {
+      clearTimeout(timeoutState);
+      setTimeoutState(undefined);
+    }
+    setTimeoutState(
+      /**
+       * These set timeouts are needed to clear the announcer string in case we have multiple
+       * participants join. Since the narrator will only announce the string in the
+       * Announcer component should the string change.
+       */
+      setTimeout(() => {
+        setAnnouncerString(string);
+        setTimeoutState(undefined);
+      }, 500)
+    );
+  };
+
+  useEffect(() => {
+    const onPersonJoined = (): void => {
+      setParticipantEventString(locale.participantJoinedNoticeString);
+    };
+    adapter.on('participantsJoined', onPersonJoined);
+
+    const onPersonLeft = (): void => {
+      setParticipantEventString(locale.participantLeftNoticeString);
```

Comment:
we should confirm expected behavior with the a11y team on what happens if someone leaves and joins, from this implementation if someone joins then before 500ms elapses someone leaves - we won't announce that someone had joined

Reply:
so the bug is explicitly because someone joined. I wonder if there is a difference for people leaving? I programmed this in because I was expecting if we announce one we should announce both. I will ping the A11y people about this.

Reply:
If you are pinging him can you ask about if multiple people joined in quick succession, and for extremes what if 50 people joined at once